meetings/working-groups/extensions/metadata-names.md

```diff
@@ -0,0 +1,183 @@
+This allows for overloading on arity while avoiding name conflicts. You can have "List" (with arity zero) and "List\`1" (with arity one).  
+For extension grouping and marker types, the name is unique enough that we use it directly as the metadata name, without adding an arity suffix.  
+But this is causing some issues.
+
+The way we produce docIDs for types is to take the name and append the arity suffix.  
+For extensions, we added special handling to use the grouping and marker names. 
+
+Let's consider this example:
+```csharp
+static class E
+{
+  extension<T>(T t)
+  {
+    public void M() { }
+  }
+}
+```
+with corresponding metadata:
+```
+.class E
+{
+  .class '<G>$8048A6C8BE30A622530249B904B537EB' // grouping type
+  {  
+    .class '<M>$65CB762EDFDF72BBC048551FDEA778ED' // marker type
+    {
+      .. marker method ..
+    }
+    public void M() => throw; // extension member without implementation
+  }
+
+  public static void M(this int i) { } // implementation method
+}
+```
+
+**If we do include an arity suffix** when producing docIDs for extensions, then the docIDs don't match the metadata names.
+
+The docIDs for the example would differ from the names in metadata:
+- extension block: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.<M>$65CB762EDFDF72BBC048551FDEA778ED\`1"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.M"
+
+
+**If we don't include an arity suffix**, then:
+- docIDs produced from VB symbols on extension metadata will diverge from those produce from C# symbols. VB doesn't have the concept of extension, so will have regular handling for types (which include an arity suffix)
+- if someone makes metadata for an extension type using some other tool, and they do include the arity suffix in the metadata names of grouping and marker types, then docIDs won't match metadata names again.
+
+The docIDs from C# source or metadata for the example would be:
+- extension block: "E.<G>$8048A6C8BE30A622530249B904B537EB.<M>$65CB762EDFDF72BBC048551FDEA778ED"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB.M"
+
+But the docIDs from VB metadata would differ from those from C#:
+- extension grouping type: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.<M>$65CB762EDFDF72BBC048551FDEA778ED\`1"
+- extension marker type: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.<M>$65CB762EDFDF72BBC048551FDEA778ED\`1"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.M"
+
+And if some other tool cooks up extension metadata including arity suffixes like this:
+```
+.class E
+{
+  .class 'GroupingName`1' // grouping type
+  {  
+    .class 'MarkerName`1' // marker type
+    {
+      .. marker method ..
+    }
+    public void M() => throw; // extension member without implementation
+  }
+
+  public static void M(this int i) { } // implementation method
+}
+```
+
+Then the docIDs would not match the metadata names:
```

Comment:
Why wouldn't they match? The types aren't generic, therefore suffixes wouldn't be added or removed by compilers. Perhaps there is a mistake in the IL and types should be generic? Suffix doesn't make types generic.

Reply:
IL got adjusted and the thread was resolved, but I am still confused by the names below. What tool/component would produce the names below? Why the arity suffix wouldn't be there? It will be part of `ExtensionGroupingName` for C# and VB would include the suffix as well. Probably in this scenario there could be a problem with a doc id that includes type parameters instead?